--- ffmpeg2theora/src/ffmpeg2theora.c 2012-01-29 09:00:19.000000000 +0100
+++ ffmpeg2theora/src/ffmpeg2theora.c 2014-07-12 21:40:10.394727058 +0200

@@ -48,6 +48,7 @@
 #include "avinfo.h"

 #define LENGTH(x) (sizeof(x) / sizeof(*x))
+#define MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio

 enum {
     NULL_FLAG,
@@ -1068,8 +1069,8 @@
         int first = 1;
         int audio_eos = 0, video_eos = 0, audio_done = 0, video_done = 0;
         int ret;
-        int16_t *audio_buf=av_malloc(4*AVCODEC_MAX_AUDIO_FRAME_SIZE);
-        int16_t *resampled=av_malloc(4*AVCODEC_MAX_AUDIO_FRAME_SIZE);
+        int16_t *audio_buf=av_malloc(4*MAX_AUDIO_FRAME_SIZE);
+        int16_t *resampled=av_malloc(4*MAX_AUDIO_FRAME_SIZE);
         int16_t *audio_p=NULL;
         int no_frames;
         int no_samples;
@@ -1530,7 +1531,7 @@
                 while((audio_eos && !audio_done) || avpkt.size > 0 ) {
                     int samples=0;
                     int samples_out=0;
-                    int data_size = 4*AVCODEC_MAX_AUDIO_FRAME_SIZE;
+                    int data_size = 4*MAX_AUDIO_FRAME_SIZE;
                     int bytes_per_sample = av_get_bytes_per_sample(aenc->sample_fmt);

                     if (avpkt.size > 0) {


